Add cudatoolkit

xmnlab opened the pull request:
Added recipe for cudatoolkit for linux.
next step would be to add recipes for win and osx platform

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/cudatoolkit```) and found it was in an excellent condition.

xmnlab commented:
added @scopatz @andersy005 as recipe-maintainers.

xmnlab commented:
it seems it is broken because some perl issue. I am taking a look on that.

scopatz commented on recipes/cudatoolkit/meta.yaml:
```diff
@@ -0,0 +1,48 @@
+  recipe-maintainers:
+    - xmnlab
+    - scopatz
+    - andersy005
```
Thanks @xmnlab! @andersy005 are you OK with being a maintainer?

andersy005 replied:
@scopatz, Yes!

scopatz commented on recipes/cudatoolkit/build.sh:
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
```
Can you add a `set -ex` lin here please, so that we can debug this? Thanks!

xmnlab replied:
sure! thanks!

scopatz commented:
Thanks for putting this in @xmnlab!

scopatz commented:
> it seems it is broken because some perl issue. I am taking a look on that.

This is probably because they used `/usr/bin/perl` in one of their scripts rather than `/usr/bin/env perl`

scopatz commented:
Also, you may be able to use a `yum_requirements.txt` file to install perl into the system.  Here is an example feedstock: https://github.com/conda-forge/imagepy-feedstock/blob/master/recipe/yum_requirements.txt  

This is an alternative to doing a sed replace on whatever file is running the perl is.

xmnlab commented:
thanks @scopatz !

I will try yum_requirements approach. normally I use `sed` to replace some code to work around some issue to build the package .. but in this case it seems that the runfile extract and run perl scripts ... I have no time to apply a `sed` command before run scrips are called.

scopatz commented:
> I have no time to apply a sed command before run scrips are called

Yeah, it would seem that you'd need to use `--extract` in order to use sed.  `yum_requirements.txt` seems to be working though.

scopatz commented:
Since this seems to be working for Linux, I'd likt o bring it to the attention of @conda-forge/staged-recipes @conda-forge/core for further discussion and review.

scopatz commented:
Ok! This looks great @xmnlab! Did you want to try to tackle Mac and or Windows here in staged-recipes or instead should we try to add them in the feedstock?

isuruf commented:
Let's keep the discussion at https://github.com/conda-forge/conda-forge.github.io/issues/63 and not merge until it is resolved

scopatz commented:
> Let's keep the discussion at conda-forge/conda-forge.github.io#63 and not merge until it is resolved

Hmmm OK.  The current discussion there doesn't seem super relate to the contents of this PR, which really just mirrors what is on defaults

xmnlab commented:
@scopatz so would be better working on here to support osx and win until https://github.com/conda-forge/conda-forge.github.io/issues/63 is resolved?

scopatz commented:
Yes @xmnlab - I think it is better to work here for the time being.

xmnlab commented:
basically there are 2 runfiles for windows:

- https://developer.nvidia.com/compute/cuda/9.2/Prod/local_installers/cuda_9.2.88_win10 (10, Server 2016)
- https://developer.nvidia.com/compute/cuda/9.2/Prod/local_installers/cuda_9.2.88_windows (8, 7, Server 2012 R2)

how to treat this case?

scopatz commented:
I think you want to use https://developer.nvidia.com/compute/cuda/9.2/Prod/local_installers/cuda_9.2.88_win10  based on this page: https://developer.nvidia.com/cuda-downloads?target_os=Windows&target_arch=x86_64&target_version=10&target_type=exelocal

The assumption should be to support Windows 10

xmnlab commented:
thanks @scopatz !

jakirkham commented:
Some relevant information in [this comment]( https://github.com/ContinuumIO/anaconda-recipes/issues/140#issuecomment-404197727 ), which may prove helpful.

jakirkham commented:
cc @jjhelmus

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/cudatoolkit```) and found some lint.

Here's what I've got...


For **recipes/cudatoolkit**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [25]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/cudatoolkit```) and found it was in an excellent condition.

xmnlab commented on recipes/cudatoolkit/bld.bat:
```diff
@@ -0,0 +1,61 @@
+    ECHO - %%f ...
+    FOR /R %install_dir% %%x IN (%%f) DO COPY "%%x" %PREFIX%\Library\include\ /Y;
+)
+
```
@jjhelmus thanks for the explanation! Yes I saw this duplication on %PREFIX%\DLLs but I didn't understand why. I will change that now :) thanks!

xmnlab commented:
[win] for some reason nvToolsExt is not inside the package after the exe is uncompressed ... maybe we will need install ...

jjhelmus commented:
> [win] for some reason nvToolsExt is not inside the package

nvToolExt is not included CUDA installer.  The [script in conda-recipe-cudatoolkit](https://github.com/numba/conda-recipe-cudatoolkit/blob/master/scripts/build.py#L45-L49) deals with this by copying the file from an existing install.

xmnlab commented:
Thanks @jjhelmus ! I will try that!! thanks a lot!

xmnlab commented:
the build for windows is working locally but it is not working on appveyor.
it seems that the installation dir was not found https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.26488#L696

maybe on appveyor the installation is located in another place.

xmnlab commented:
[osx]

```

FileNotFoundError: [Errno 2] No such file or directory: '/Users/travis/miniconda/conda-bld/cudatoolkit_1532122534164/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/libcublas.9.2.dylib'
```

it seems this file was not found by `find` command

```
+ echo '- libcublas ...'

- libcublas ...

+ find /Users/travis/miniconda/conda-bld/cudatoolkit_1532122534164/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/tmp/cuda_install -name 'libcublas*' -exec cp -a '{}' /Users/travis/miniconda/conda-bld/cudatoolkit_1532122534164/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib ';'

cp: symlink: /Developer/NVIDIA/CUDA-9.2/lib/libcublas.dylib: File exists
```

xmnlab commented:
hi @scopatz 

this is the comment about current win packaging status: https://github.com/conda-forge/staged-recipes/pull/6240#issuecomment-406743808

and thsi is the comment about the current osx packaging status:  https://github.com/conda-forge/staged-recipes/pull/6240#issuecomment-406744502

scopatz commented:
@xmnlab - hmm I am not sure why find isn't working on Mac? Have you tried this on our test machine yet?

For windows, it seems to be failing at the test stage.  Are we sure that this numba-based test is correct for this platform?

xmnlab commented:
@scopatz 

[osx]
I have tested the `find` command on an osx and it has the same behavior of the `find` command of linux. So I think it could be some issues related to the extraction and/or the paths used. I have not tested on our test machine. I am going do it to figure out better the paths.

[win]
I think the error: ```numba.cuda.cudadrv.error.NvvmSupportError: libNVVM cannot be found. Do `conda install cudatoolkit`: library nvvm not found``` is related to a fail on copy files because there is no files on the package: https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.26488#L1212

scopatz commented:
[win] - right but it does seem like the `nvvm*.dll` is getting copied over.  Or maybe it isn't because the `*` isn't being expanded.  Have you run that locally?

ocefpaf commented:
@xmnlab and @scopatz is Linux good to go? Are we OK merging this toolkit?
If yes to both do you want to merge this as-is and work on OS X and Windows on the feedstock?

scopatz commented:
@ocefpaf - yeah, linux is good to go! I would be happy working on the mac and windows builds in the feedstock.

@isuruf said he didn't want this merged until https://github.com/conda-forge/conda-forge.github.io/issues/63 was resolved, but I am not sure what exactly he means by that at this point.  @isuruf - do you feel that issue has been resolved?

isuruf commented:
No. I remember discussions where 'defend and indemnify' the software author came up, and in all of those, it was decided not a good idea.

https://github.com/conda-forge/staged-recipes/issues/80#issuecomment-209558884
https://github.com/conda/conda/issues/5383#issuecomment-303015020
https://github.com/conda-forge/staged-recipes/issues/3022#issuecomment-305882455

scopatz commented:
@isuruf - those are all releated to the MKL and not cudatoolkit. The EULA are totally different and not really comparable.

scopatz commented:
Also, as stated I believe that our BSD license protects conda-forge from the cudatoolkit EULA defend and indemnify clause. That was never responded to in conda-forge/conda-forge.github.io#63

scopatz commented:
I think I see the issue with the mac package.  I added the middle find line:

```
    echo "- $f ..."
    find $install_dir -name "${f}*"  -exec ls -l {} \;
    find $install_dir -name "${f}*"  -exec cp -a {} $PREFIX/lib \;
```

This provides output like:

```
- libcublas ...
+ for f in '$cuda_libs'
+ echo '- libcublas ...'
+ find /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install -name 'libcublas*' -exec ls -l '{}' ';'
-rw-r--r--  1 anthony  staff  65699984 Jun 13 10:35 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/Developer/NVIDIA/CUDA-9.2/lib/libcublas_device.a
-rwxr-xr-x  1 anthony  staff  54484096 Jun 13 10:35 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/Developer/NVIDIA/CUDA-9.2/lib/libcublas.9.2.dylib
-rw-r--r--  1 anthony  staff  65049352 Jun 13 10:35 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/Developer/NVIDIA/CUDA-9.2/lib/libcublas_static.a
lrwxr-xr-x  1 anthony  staff  19 Jun 13 10:35 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/Developer/NVIDIA/CUDA-9.2/lib/libcublas.dylib -> libcublas.9.2.dylib
-rw-r--r--  1 anthony  staff  26 Jun 13 10:35 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/Developer/NVIDIA/CUDA-9.2/doc/man/man7/libcublas.7
-rw-r--r--  1 anthony  staff  26 Jun 13 10:35 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/Developer/NVIDIA/CUDA-9.2/doc/man/man7/libcublas.so.7
lrwxr-xr-x  1 anthony  staff  49 Jun 13 10:36 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/usr/local/cuda/lib/libcublas_device.a -> /Developer/NVIDIA/CUDA-9.2/lib/libcublas_device.a
lrwxr-xr-x  1 anthony  staff  50 Jun 13 10:36 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/usr/local/cuda/lib/libcublas.9.2.dylib -> /Developer/NVIDIA/CUDA-9.2/lib/libcublas.9.2.dylib
lrwxr-xr-x  1 anthony  staff  49 Jun 13 10:36 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/usr/local/cuda/lib/libcublas_static.a -> /Developer/NVIDIA/CUDA-9.2/lib/libcublas_static.a
lrwxr-xr-x  1 anthony  staff  46 Jun 13 10:36 /Users/anthony/miniconda/conda-bld/cudatoolkit_1533052794633/_build_env/tmp/cuda_install/usr/local/cuda/lib/libcublas.dylib -> /Developer/NVIDIA/CUDA-9.2/lib/libcublas.dylib
```

So this implies that `libcublas.9.2.dylib` is being overwritten by a soft-link. The real file (in `cuda_install/Developer/NVIDIA/CUDA-9.2/lib`) is being overwritten by a soft-link file later on (in `cuda_install/usr/local/cuda/lib`). 

I think we only want the one in the `Developer/` dir

isuruf commented:
> @isuruf - those are all releated to the MKL and not cudatoolkit. The EULA are totally different and not really comparable.

How is it different? Both have similar clauses regarding indemnification.

> Also, as stated I believe that our BSD license protects conda-forge from the cudatoolkit EULA defend and indemnify clause. That was never responded to in conda-forge/conda-forge.github.io#63

IANAL, so I don't if it's true or not.

scopatz commented:
> How is it different? Both have similar clauses regarding indemnification.

I am not intimately familiar with the MKL license, but the context of such statements is important. For example, both GPL and BSD include "no warranty, as-is" clauses, but we should hardly treat these as the same license.

> IANAL, so I don't if it's true or not.

If that is the issue, we can get access to legal council through NumFOCUS. In preparation for that, though, it would be helpful to know what exactly the questions or concerns that we have are. What questions would you actually like to see answered?

scopatz commented on recipes/cudatoolkit/build.sh:
```diff
@@ -0,0 +1,88 @@
+    chmod ugo+x $filename
+    ./$filename --silent --toolkit --toolkitpath=$install_dir --override
+else
+    find_args="-type f"
```
I think you want `-xtype f` here instead, so it doesn't pick up symbolic links.

xmnlab replied:
@scopatz thanks .. I will try that ... In our server it is already working ...

xmnlab replied:
the output of the building on our server:

```
===== testing package: cudatoolkit-9.2-0 =====
running run_test.py
Finding cublas
        located at /Users/Shared/xmnlab/conda-forge/miniconda3/conda-bld/cudatoolkit_1533086615862/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcublas.9.2.dylib
        trying to open library...       ok
Finding cusparse
        located at /Users/Shared/xmnlab/conda-forge/miniconda3/conda-bld/cudatoolkit_1533086615862/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcusparse.9.2.dylib
        trying to open library...       ok
Finding cufft
        located at /Users/Shared/xmnlab/conda-forge/miniconda3/conda-bld/cudatoolkit_1533086615862/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcufft.9.2.dylib
        trying to open library...       ok
Finding curand
        located at /Users/Shared/xmnlab/conda-forge/miniconda3/conda-bld/cudatoolkit_1533086615862/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcurand.9.2.dylib
        trying to open library...       ok
Finding nvvm
        located at /Users/Shared/xmnlab/conda-forge/miniconda3/conda-bld/cudatoolkit_1533086615862/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libnvvm.3.dylib
        trying to open library...       ok
        finding libdevice for compute_20...     ok
        finding libdevice for compute_30...     ok
        finding libdevice for compute_35...     ok
        finding libdevice for compute_50...     ok
NVVM version (1, 5)

Resource usage statistics from testing cudatoolkit:
   Process count: 2
   CPU time: Sys=0:00:00.3, User=0:00:00.4
   Memory: 47.8M
   Disk usage: 24B
   Time elapsed: 0:00:04.2

TEST END: /Users/Shared/xmnlab/conda-forge/miniconda3/conda-bld/osx-64/cudatoolkit-9.2-0.tar.bz2
```

xmnlab replied:
@scopatz it seems xtype is not known: `find: -xtype: unknown primary or operator`

this is the info about the -type parameter:

``` 
 -type t
             True if the file is of the specified type.  Possible file types are as follows:

             b       block special
             c       character special
             d       directory
             f       regular file
             l       symbolic link
             p       FIFO
             s       socket

```

xmnlab commented:
refs for test on windows: https://github.com/numba/conda-recipe-cudatoolkit/pull/9/files#diff-8c4bbdb6bc02365381f54967559b8af0R8

xmnlab commented:
not sure if it could be related to this problem: 

```
1.1. System Requirements
To use CUDA on your system, you need to have:
a CUDA-capable GPU
Mac OS X 10.13
the Clang compiler and toolchain installed using Xcode
the NVIDIA CUDA Toolkit (available from the CUDA Download page)
```
source: https://docs.nvidia.com/cuda/cuda-installation-guide-mac-os-x/index.html

it seems that osx version of conda-forge is lower:

```
+ sw_vers

ProductName:	Mac OS X

ProductVersion:	10.10.5

BuildVersion:	14F1808
```

xmnlab commented:
I fixed the command to remove unnecessary folder before copy files to $PREFIX. 
it seems now the files are right on the package. but it is still failing.
I changed again to just copy to PREFIX regular files (-type f) ... just waiting now for the results on travis-ci

xmnlab commented:
it didn't work .. I will compare the result of this package with https://anaconda.org/anaconda/cudatoolkit/9.0/download/osx-64/cudatoolkit-9.0-h41a26b3_0.tar.bz2

xmnlab commented:
current error message:

```
===== testing package: cudatoolkit-9.2-0 =====

running run_test.py

Finding cublas

	located at /Users/travis/miniconda/conda-bld/cudatoolkit_1533255702942/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcublas.9.2.dylib

	trying to open library...	ERROR: failed to open cublas:

dlopen(/Users/travis/miniconda/conda-bld/cudatoolkit_1533255702942/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcublas.9.2.dylib, 6): no suitable image found.  Did find:

	/Users/travis/miniconda/conda-bld/cudatoolkit_1533255702942/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcublas.9.2.dylib: code signature invalid for '/Users/travis/miniconda/conda-bld/cudatoolkit_1533255702942/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcublas.9.2.dylib'



Finding cusparse

	located at /Users/travis/miniconda/conda-bld/cudatoolkit_1533255702942/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcusparse.9.2.dylib

Finding cufft

	located at /Users/travis/miniconda/conda-bld/cudatoolkit_1533255702942/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcufft.9.2.dylib

Finding curand

	located at /Users/travis/miniconda/conda-bld/cudatoolkit_1533255702942/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libcurand.9.2.dylib

Finding nvvm

	located at /Users/travis/miniconda/conda-bld/cudatoolkit_1533255702942/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/libnvvm.3.dylib

	finding libdevice for compute_20...	ok

	finding libdevice for compute_30...	ok

	finding libdevice for compute_35...	ok

	finding libdevice for compute_50...	ok

Tests failed for cudatoolkit-9.2-0.tar.bz2 - moving package to /Users/travis/miniconda/conda-bld/broken

WARNING:conda_build.build:Tests failed for cudatoolkit-9.2-0.tar.bz2 - moving package to /Users/travis/miniconda/conda-bld/broken

removing: cudatoolkit-9.2-0.tar.bz2

TESTS FAILED: cudatoolkit-9.2-0.tar.bz2
```
any recommendation or advice is very welcome :)

scopatz commented:
@xmnlab - wow - that is a strange error. Is that the same error as on the build machine we have?

scopatz commented:
As discussed privately, those test errors may be because cudatoolkit doesn't work with older versions of the macOS SDK. Is it possible to bump the SDK number for just a particular package? Does anyone on @conda-forge/staged-recipes or @conda-forge/core know?

xmnlab commented:
it seems there are 2 libraries that are needed. I don't know if it cold be the problem here:

- Needed DSO /usr/lib/libncurses.5.4.dylib
- Needed DSO /usr/lib/libc++.1.dylib

scopatz commented:
Ahh OK, that is pretty easy to solve, I think you need to add the following dependncies:

```
build:
  - {{ compiler('cxx') }}   # [osx]
host:
  - ncurses  # [osx]
```

scopatz commented:
made a slight edit to add run dep.

xmnlab commented:
great! thanks @scopatz ! I am going to add that!

xmnlab commented:
[win] for some reason there is no "NVIDIA GPU Computing Toolkit" folder [1] .. I will mix installation and extraction approach. 

[1] https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.26980#L755

xmnlab commented:
@scopatz I changed the tests for osx to check the files there.

karl616 commented:
Could it have something to do with DYLD_LIBRARY_PATH? Wild guess and I found this in the documentation:

> Set up the required environment variables:
> ```
> export PATH=/Developer/NVIDIA/CUDA-9.2/bin
> ${PATH:+:${PATH}}
> export DYLD_LIBRARY_PATH=/Developer/NVIDIA/CUDA-9.2/lib\
>                          ${DYLD_LIBRARY_PATH:+:${DYLD_LIBRARY_PATH}
> ```

It is a wild guess, the path above is probably wrong... and I thought conda normally handled things like this on it's own. 

Perhaps add an `echo ${DYLD_LIBRARY_PATH} to the build script... Hope someone with more osx experience know better...

xmnlab commented:
thanks @karl616 for the feedback! 

we probably should avoid to change DYLD_LIBRARY_PATH to avoid conflicts with conda. but it seems to be a good point and we should have this in mind. Thanks so much!

scopatz commented:
This is now passing on all platforms - and is ready for review by @conda-forge/staged-recipes!

jjhelmus commented:
This looks to be including both the dynamic and static libraries (at least on linux) which results in a really large package, ~770 MB vs 350 MB for just the dynamic libraries.  Is there a reason for including the static libraries?  Would it be reasonable to split these into their only package, cudatoolkit-static?

scopatz commented:
This is a great suggestion @jjhelmus! This package should include only the dynamic libraries.

hadim commented:
Any chance to get this merged?

hadim commented:
Ok, I just realized there is some licensing issue at https://github.com/conda-forge/conda-forge.github.io/issues/63.

leofang commented:
just curious, is this PR superseded by something else or abandoned?

andersy005 commented:
@leofang 

> just curious, is this PR superseded by something else or abandoned?

https://github.com/conda-forge/cudatoolkit-dev-feedstock

xmnlab commented:
@scopatz should we keep it opened? or should we close it?

scopatz commented:
It would be nice to have this still, so we don't rely on defaults, but we need permission from Nvidia.
